tuned/plugins/plugin_scheduler.py

```diff
@@ -350,8 +410,103 @@ def _convert_sched_cfg(self, vals):
 		affinity = self._convert_affinity(affinity)
 		return (rule_prio, scheduler, priority, affinity, regex)
 
+	def _cgroup_create_group(self, cgroup):
+		path = "%s/%s" % (self._cgroup_mount_point, cgroup)
+		try:
+			os.mkdir(path, consts.DEF_CGROUP_MODE)
+		except OSError as e:
+			log.error("Unable to create cgroup '%s': %s" % (path, e))
+		if (not self._cmd.write_to_file("%s/%s" % (path, "cpuset.mems"),
+				self._cmd.read_file("%s/%s" % (self._cgroup_mount_point, "cpuset.mems"), no_error = True),
+				no_error = True)):
+					log.error("Unable to initialize 'cpuset.mems ' for cgroup '%s'" % path)
+		if (not self._cmd.write_to_file("%s/%s" % (path, "cpuset.cpus"),
+				self._cmd.read_file("%s/%s" % (self._cgroup_mount_point, "cpuset.cpus"), no_error = True),
+				no_error = True)):
+					log.error("Unable to initialize 'cpuset.cpus' for cgroup '%s'" % path)
+
+	def _cgroup_initialize_groups(self):
+		if self._cgroup is not None and not self._cgroup in self._cgroups:
+			self._cgroup_create_group(self._cgroup)
+		for cg in self._cgroups:
+			self._cgroup_create_group(cg)
+
+	def _cgroup_initialize(self):
+		log.debug("Initializing cgroups settings")
+		try:
+			os.makedirs(self._cgroup_mount_point, consts.DEF_CGROUP_MODE)
+		except OSError as e:
+			log.error("Unable to create cgroup mount point: %s" % e)
+		(ret, out) = self._cmd.execute(["mount", "-t", "cgroup", "-o", "cpuset", "cpuset", self._cgroup_mount_point])
+		if ret != 0:
+			log.error("Unable to mount '%s'" % self._cgroup_mount_point)
+
+	def _remove_dir(self, cgroup):
+		try:
+			os.rmdir(cgroup)
+		except OSError as e:
+			log.error("Unable to remove directory '%s': %s" % (cgroup, e))
+
+	def _cgroup_finalize_groups(self):
+		if self._cgroup is not None and not self._cgroup in self._cgroups:
+			self._remove_dir("%s/%s" % (self._cgroup_mount_point, self._cgroup))
+		for cg in self._cgroups:
```

Comment:
I think the directories need to be removed in reversed order. If you have `cgroup.parent` and `cgroup.parent/child`, then removing parent will fail because child is not yet removed at that point.

Reply:
Reversed.